Demo/Fullscreen/FullscreenGallery/FullscreenGalleryDemoViewController.swift

```diff
@@ -0,0 +1,304 @@
+        playgroundView.addSubview(thumbnailLabel)
+        playgroundView.addSubview(simulateLoadingSwitch)
+        playgroundView.addSubview(simulateLoadingLabel)
+        playgroundView.addSubview(helpLabel)
+
+        NSLayoutConstraint.activate([
+            collectionView.heightAnchor.constraint(equalToConstant: collectionCellHeight + CGFloat.largeSpacing),
+            collectionView.leadingAnchor.constraint(equalTo: playgroundView.leadingAnchor),
+            collectionView.trailingAnchor.constraint(equalTo: playgroundView.trailingAnchor),
+            collectionView.centerYAnchor.constraint(equalTo: playgroundView.centerYAnchor),
+
+            thumbnailSwitch.topAnchor.constraint(equalTo: collectionView.bottomAnchor, constant: .largeSpacing),
+            thumbnailSwitch.leadingAnchor.constraint(equalTo: playgroundView.leadingAnchor, constant: .largeSpacing),
+
+            thumbnailLabel.leadingAnchor.constraint(equalTo: thumbnailSwitch.trailingAnchor, constant: .mediumLargeSpacing),
+            thumbnailLabel.centerYAnchor.constraint(equalTo: thumbnailSwitch.centerYAnchor),
+
+            simulateLoadingSwitch.topAnchor.constraint(equalTo: thumbnailSwitch.bottomAnchor, constant: .mediumSpacing),
+            simulateLoadingSwitch.leadingAnchor.constraint(equalTo: playgroundView.leadingAnchor, constant: .largeSpacing),
+
+            simulateLoadingLabel.leadingAnchor.constraint(equalTo: simulateLoadingSwitch.trailingAnchor, constant: .mediumLargeSpacing),
+            simulateLoadingLabel.centerYAnchor.constraint(equalTo: simulateLoadingSwitch.centerYAnchor),
+
+            helpLabel.leadingAnchor.constraint(equalTo: playgroundView.leadingAnchor, constant: .mediumLargeSpacing),
+            helpLabel.trailingAnchor.constraint(equalTo: playgroundView.trailingAnchor, constant: -.mediumLargeSpacing),
+            helpLabel.topAnchor.constraint(equalTo: playgroundView.safeLayoutGuide.topAnchor, constant: .largeSpacing),
+
+        ])
+    }
+
+    // MARK: - Private methods
+
+    private func displayFullscreenGallery(forIndex index: Int) {
+        let viewModel = DemoViewModel(imageUrls: imageUrls, imageCaptions: imageCaptions, selectedIndex: index)
+
+        let gallery = FullscreenGalleryViewController(viewModel: viewModel, thumbnailsInitiallyVisible: thumbnailSwitch.isOn)
+        gallery.galleryDelegate = self
+        gallery.galleryDataSource = ImageDownloader.shared
+        gallery.transitioningDelegate = transitionController
+
+        present(gallery, animated: true)
+    }
+
+    @objc private func loadSimulationSwitchToggled() {
+        if simulateLoadingSwitch.isOn {
+            ImageDownloader.shared.simulatedDelayMs = 200
+        } else {
+            ImageDownloader.shared.simulatedDelayMs = 0
+        }
+    }
+
+    // MARK: - UICollectionView
+
+    public func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 5
```

Comment:
Nitpicking, but could potentially be `min(imageUrls.count, imageCaptions.count)` instead.